tests/login.rs

```diff
@@ -16,9 +16,16 @@ use cargo::core::Shell;
 use hamcrest::{assert_that, existing_file, is_not};
 
 const TOKEN: &str = "test-token";
+const ORIGINAL_TOKEN: &str = "api-token";
 const CONFIG_FILE: &str = r#"
     [registry]
     token = "api-token"
+
+    [registries.test-reg]
+    index = "http://dummy_index/"
+
+    [registries.test.reg]
```

Comment:
This actually reminds me, at some point we're going to want validation on the names of registries I think? Probably for now (to start conservative at least) we probably want to restrict registries to `[a-zA-Z0-9_-]` probably?

Reply:
Restrict the identifiers of the registry, that is

Reply:
In that case I will remove the test for test.reg and remove this entry. I will also add this to the issue list for stabilization.